src/metabase/query_processor/middleware/large_int_id.clj

```diff
@@ -15,28 +15,33 @@
           (reduce #(update (vec %1) %2 ->string) row field-indexes)))
    rf))
 
-(defn- should-convert-to-string? [field]
-  (and (or (isa? (:semantic-type field) :type/PK)
-           (isa? (:semantic-type field) :type/FK))
-       (or (isa? (:base-type field) :type/Integer)
-           (isa? (:base-type field) :type/Number))))
+(defn- should-convert-to-string?
+  "Determines if values of the given column should be converted to strings for JavaScript safety.
 
-(defn- field-indexes [fields]
-  (not-empty
-   (keep-indexed
-    (fn [idx val]
-      ;; TODO -- we could probably fix the rest of #5816 by adding support for
-      ;; `:field` w/ name and removing the PK/FK requirements -- might break
-      ;; the FE client tho.
-      (when-let [field (lib.util.match/match-one val
-                         [:field (field-id :guard integer?) _]
-                         ;; TODO -- can't we use the QP store here? Seems like
-                         ;; we should be able to, but it doesn't work (not
-                         ;; initialized)
-                         (lib.metadata.protocols/field (qp.store/metadata-provider) field-id))]
-        (when (should-convert-to-string? field)
-          idx)))
-    fields)))
+  PKs and FKs which are numbers need converting, lest they lose precision. JS numbers only guarantee 52 bits."
+  [{:keys [base-type semantic-type] :as _column-metadata}]
+  (and (or (isa? semantic-type :type/PK)
+           (isa? semantic-type :type/FK))
+       (or (isa? base-type     :type/Integer)
+           (isa? base-type     :type/Number))))
+
+(defn- field-indexes [query]
```

Comment:
Even in this namespace, I think this name is too general. (I understand this is not new in this PR.)

Reply:
This actually got reverted anyway, due to a concurrent change in this file. It was no longer needed.